components/proxy/src/main/java/com/hotels/styx/routing/handlers/RouteRefLookup.java

```diff
@@ -31,20 +29,28 @@
 /**
  * Resolves a routing object reference from route database.
  */
-public class RouteRefLookup implements Function<RoutingObjectReference, HttpHandler> {
-    private final StyxObjectStore<RoutingObjectRecord> routeDatabase;
-
-    public RouteRefLookup(StyxObjectStore<RoutingObjectRecord> handler) {
-        this.routeDatabase = requireNonNull(handler);
-    }
-
-    public HttpHandler apply(RoutingObjectReference route) {
-        return this.routeDatabase.get(route.name())
-                .map(RoutingObjectRecord::getHandler)
-                .orElse((x, y) -> Eventual.of(response(NOT_FOUND)
-                        .body("Not found: " + route.name(), UTF_8)
-                        .build()
-                        .stream()
-                ));
+public interface RouteRefLookup {
+    HttpHandler apply(RoutingObjectReference route);
+
+    /**
+     * A StyxObjectStore based route reference lookup function.
+     */
+    class RouteDbRefLookup implements RouteRefLookup {
+        private final StyxObjectStore<RoutingObjectRecord> routeDatabase;
+
+        public RouteDbRefLookup(StyxObjectStore<RoutingObjectRecord> routeDatabase) {
+            this.routeDatabase = requireNonNull(routeDatabase);
+        }
+
+        @Override
+        public HttpHandler apply(RoutingObjectReference route) {
+            return this.routeDatabase.get(route.name())
+                    .map(RoutingObjectRecord::getHandler)
+                    .orElse((x, y) -> Eventual.of(response(NOT_FOUND)
```

Comment:
During discussion we noticed that there may be a bug here and in many similar circumstances in which a `LiveHttpRequest` is not having its body consumed.

Reply:
Fixed this handler. However I will review the usage of other handlers separately.